proxy-support/README.md

```diff
@@ -0,0 +1,273 @@
+    httpsProxy: http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128
+```
+
+This `secret` and `configmap` can now be used to pass these information into `Apps` which get applied into a WC.
+As `cert-manager` needs this data, the manifest should now look like:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+[...]
+spec:
+  catalog: default-test
+  config:
+    configMap:
+      name: mycluster-cluster-values
+      namespace: org-giantswarm
+    secret:
+      name: mycluster-cluster-values
+      namespace: org-giantswarm
+[...]
+```
+
+As we have these information on a cluster-base, we are now able to create a new `secret`, called `<clusterName>-systemd-proxy` which is a rendered version of [this](#containerd) systemd drop-in. In our example, the `secret` now looks like
+
+```yaml
+[Service]
+Environment="HTTP_PROXY=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="http_proxy=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="HTTPS_PROXY=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="https_proxy=http://myProxyUser:mySuperSecretPassword@192.168.52.220:3128"
+Environment="NO_PROXY=172.16.0.0/16,internal.corpdomain.net,example.com"
+Environment="no_proxy=172.16.0.0/16,internal.corpdomain.net,example.com"
+```
+
+This secret can now be referenced in a `kubeadmconfig`, e.g.:
+
+```yaml
+files:
+  - path: /etc/systemd/system/containerd.service.d/http-proxy.conf
+    permissions: "0600"
+    contentFrom:
+      secret:
+        name: <clusterName>-systemd-proxy
+        key: containerdProxy   
+```
+
+By doing this, `ClusterAPI` is waiting for the `secret` to get created. Once the `secret` got created, it will be used by creating the machines (and injecting this data via `cloud-init` or `ignition`).
+
+As `cluster-apps-operator` takes care of creating `deployment/chart-operator` in the WC with the values from `(secret|configmap)/<clusterName>-cluster-values` as input, `chart-operator` in every WC automatically get a valid proxy configuration.
+
+## To be discussed
+
+- Customers are able to see the proxy configuration (especially the secret part) within a `WC`. Is this a security issue?
+- Should the [`cluster-apps-operator`] known installation wide proxy configuration propagated into WCs for [`kyverno-policies-connectivity`]?
+- Should a customer defined Proxy configuration used for `containerd`, `cert-manager` and `chart-operator` as well or only for customer workload?
+
+## Current known limitations and issues:
+
+- If customers using a service mesh (e.g. `linkerd`) and the mesh is spawn over multiple WCs, the current implementation provides that all mesh specific member values are defined in the global proxy configuration.
+- If a customer want a specific proxy configuration for a workload cluster, this proxy configuration must be currently defined in multiple places.
```

Comment:
At the moment it's technically possible to overwrite the proxyConfiguration which is required for the initial cluster bootstraping (proxy configuration in `containerd` and `certManager`) but this has to be done on multiple places (`secret` for `containerd` must be created, `values.yaml` for `certManager` must be configured)

We could think about extending `cluster-apps-operator` by propagating a custom proxy configuration to the workload clusters if the workload cluster specific configuration got created in a specific way (tbd e.g. a `secret` or `configmap` with a fixed naming pattern) beforehand.